dirty_cat/datasets/fetching.py

```diff
@@ -1,493 +1,396 @@
 
-    data = fetch_openml(data_id=42125, as_frame=True)
-    data.data['Current Annual Salary'] = data['target']
-    return data
-
-    # link dead.
-    # return fetch_dataset(EMPLOYEE_SALARIES_CONFIG, show_progress=False)
 
+def fetch_openml_dataset(dataset_id: int,
+                         data_directory: Path = get_data_dir()) -> dict:
+    """
+    Gets a dataset from OpenML (https://www.openml.org),
+    or from the disk if already downloaded.
 
-def fetch_road_safety():
-    """fetches the road safety dataset
+    Parameters
+    ----------
+    dataset_id: int
+        The ID of the dataset to fetch.
+    data_directory: Path
+        Optional. A directory to save the data to.
+        By default, the dirty_cat data directory.
 
     Returns
     -------
     dict
-        a dictionary containing:
-
-            - a short description of the dataset (under the ``description``
-              key)
-            - an absolute path leading to the csv file where the data is stored
-              locally (under the ``path`` key)
-
+        A dictionary containing:
+          - ``description``: str
+              The description of the dataset,
+              as gathered from OpenML.
+          - ``source``: str
+              The dataset's URL from OpenML.
+          - ``path``: pathlib.Path
+              The local path leading to the dataset,
+              saved as a CSV file.
+
+          The following values are added by the fetches below (fetch_*)
+          - ``read_csv_kwargs``: Dict[str, Any]
+              A list of keyword arguments that can be passed to
+              `pandas.read_csv` for reading.
+              Usually, it contains `quotechar`, `escapechar` and `na_values`.
+              See `pandas.read_csv`'s documentation for more information.
+              Use by passing `**info['read_csv_kwargs']` to `read_csv`.
+          - ``y``: str
+              The name of the target column.
 
-    References
-    ----------
-    https://data.gov.uk/dataset/road-accidents-safety-dataset
     """
 
-    return fetch_dataset(ROAD_SAFETY_CONFIG, show_progress=False)
-
-
-def fetch_medical_charge():
-    """fetches the medical charge dataset
-
-    Returns
-    -------
-    dict
-        a dictionary containing:
+    # Construct the path to the gzip file containing the details on a dataset.
+    details_gz_path = data_directory / DETAILS_DIRECTORY / f'{dataset_id}.gz'
+    features_gz_path = data_directory / FEATURES_DIRECTORY / f'{dataset_id}.gz'
+
+    if not details_gz_path.is_file() or not features_gz_path.is_file():
+        # If the details file or the features file don't exist,
+        # download the dataset.
+        warnings.warn(
+            f"Could not find the dataset {dataset_id} locally. "
+            "Downloading it from OpenML; this might take a while... "
+            "If the process is interrupted, files will be invalid/incomplete. "
+            "To fix this problem, delete the CSV file if it exists. "
```

Comment:
Maybe we should print also the path of the CSV file to help the user

Reply:
You're right ! I don't like this message altogether, it's too vague and inaccurate, I'll change it.